url7b90c1010b_DastanIqbal_HelloZMQwithJava_tgz-pJ8-ClientJ8/src/zmq/Mtrie.java

```diff
@@ -0,0 +1,394 @@
+/*              
+    Copyright (c) 2011 250bpm s.r.o.
+    Copyright (c) 2011-2012 Spotify AB
+    Copyright (c) 2011 Other contributors as noted in the AUTHORS file
+                    
+    This file is part of 0MQ.
+                            
+    0MQ is free software; you can redistribute it and/or modify it under
+    the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    0MQ is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+                
+    You should have received a copy of the GNU Lesser General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/ 
+package zmq;
+
+import java.util.HashSet;
+import java.util.Set;
+
+//Multi-trie. Each node in the trie is a set of pointers to pipes.
+public class Mtrie {
+    private Set<Pipe> pipes;
+
+    private int min;
+    private int count;
+    private int live_nodes;
+    private Mtrie[] next;
+    
+    public static interface IMtrieHandler {
+        void invoke(Pipe pipe, byte[] data, Object arg);
+    }
+    
+    public Mtrie() {
+        min = 0;
+        count = 0;
+        live_nodes = 0;
+        
+        pipes = null;
+        next = null;
+    }
+    
+    public boolean add (byte[] prefix_, Pipe pipe_)
+    {
+        return add_helper (prefix_, 0, pipe_);
+    }
+
+    //  Add key to the trie. Returns true if it's a new subscription
+    //  rather than a duplicate.
+    public boolean add (byte[] prefix_, int start_, Pipe pipe_)
+    {
+        return add_helper (prefix_, start_, pipe_);
+    }
+
+
+    private boolean add_helper (byte[] prefix_, int start_, Pipe pipe_)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 64-70 into a separate method.